pandas/core/frame.py

```diff
@@ -3437,6 +3437,74 @@ def f(vals):
 
     # ----------------------------------------------------------------------
     # Sorting
+    def _get_column_or_level_values(self, key, axis=1,
+                                    op_description='retrieve'):
+        """
+        Return an array of values from a column or index level.
+
+        Parameters
+        ----------
+        key: int or object
```

Comment:
This should *only* accept string keys referring to a named column or level, We do not want to have the integer/string mess like ``.ix`` again.